### PR TITLE
Support interface for attached_disk

### PR DESCRIPTION
**Pull request by chentaozgoogle**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `interface` field to `google_compute_attached_disk`
```

### CODE REVIEW

**Review by slevenick — COMMENTED** [review]
We specifically removed this field a few years ago: https://github.com/GoogleCloudPlatform/magic-modules/pull/5185

Why would it be useful now?

**Comment by chentaozgoogle** [comment]
@slevenick@
 We need to  submit this PR and https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/10591012636/job/29347715483 to support inteface for our customers.

I made the logic of ignoring   'interface' from google cloud when the interface is NOT configured in terraform. I expect this can prevent breaking customers as currently no customers are using the field. Let me know if you have cocnern or not.

**Comment by chentaozgoogle** [comment]
> Tests passed during RECORDING mode: `TestAccComputeInstanceTemplate_withNamePrefix`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11286/artifacts/5787b79f-dc28-47be-98db-f4561d481438/recording/TestAccComputeInstanceTemplate_withNamePrefix.log)] Tests failed when rerunning REPLAYING mode: `TestAccComputeInstanceTemplate_withNamePrefix`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11286/artifacts/5787b79f-dc28-47be-98db-f4561d481438/build-log/replaying_build_after_recording/TestAccComputeInstanceTemplate_withNamePrefix_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11286/artifacts/5787b79f-dc28-47be-98db-f4561d481438/replaying_after_recording/TestAccComputeInstanceTemplate_withNamePrefix.log)]
> 
> Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.
> 
> Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.
> 
> All tests passed!
> 
> View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11286/artifacts/5787b79f-dc28-47be-98db-f4561d481438/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-11286/artifacts/5787b79f-dc28-47be-98db-f4561d481438/recording) for each test

The failure should not be related to this PR

File: mmv1/third_party/terraform/services/compute/resource_compute_attached_disk_test.go [hunk 1]
```diff
@@ -315,3 +315,56 @@ resource "google_compute_attached_disk" "test" {
...
+				Config: testAttachedDiskResource(diskName, instanceName) + testAccComputeAttachedDisk_interface("testInterface", "SCSI"),
+			},
+			{
+				ResourceName:      "google_compute_attached_disk.testInterface",
+				ImportStateId:     importID,
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+			{
+				Config: testAttachedDiskResource(diskName, instanceName) + testAccComputeAttachedDisk_noInterface("testNoInterface"),
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/compute/resource_compute_attached_disk_test.go:340]: Can you add a test step that adds an interface where there was not one specified? And one that changes the interface on a disk?

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/compute/resource_compute_attached_disk_test.go (hunk 1 above)
Comment (chentaozgoogle) [reply mmv1/third_party/terraform/services/compute/resource_compute_attached_disk_test.go]: made the field foreNew and ignore read and added tests

File: mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb [hunk 2]
```diff
@@ -87,6 +87,14 @@ func ResourceComputeAttachedDisk() *schema.Resource {
 				Description:  `The mode in which to attach this disk, either READ_WRITE or READ_ONLY. If not specified, the default is to attach the disk in READ_WRITE mode.`,
 				ValidateFunc: validation.StringInSlice([]string{"READ_ONLY", "READ_WRITE"}, false),
 			},
+                        "interface": {
+                                Type:         schema.TypeString,
+                                ForceNew:     true,
+                                Optional:     true,
+                                Description:  `The disk interface used for attaching this disk. One of SCSI or NVME. (This field is only used for specific cases, please don't specify this field without advice from Google.)`,
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb:94]: You'll need to add this to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/compute_attached_disk.html.markdown as well

**Review by slevenick — CHANGES_REQUESTED** [review]
Should we document something like, NVME is the only applicable interface for most situations

File: mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb (hunk 2 above)
Comment (chentaozgoogle) [reply mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb:94]: Can we not expose this in the public doc right now? We don't want to advertise this feature at this moment

Comment (chentaozgoogle) [reply mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb:94]: Actually NVME is not for most situations. Most gen2 machines or older will be scsi and newer machines will NVME , but it still  be case by case

Comment (slevenick) [reply mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb:94]: No, we need to document every field that is available. If we don't want it to be public we shouldn't release it in the provider. We can hold off on merging until we are ready to advertise the feature.

File: mmv1/third_party/terraform/website/docs/r/compute_attached_disk.html.markdown [hunk 3]
```diff
@@ -101,6 +101,18 @@ The following arguments are supported:
 	  "READ_ONLY"
 	  "READ_WRITE"
 
+* `interface` -
+  (Optional)
+  The disk interface used for attaching this disk. One of SCSI or NVME.
+
+	This field is only used for specific cases, please don't specify
+	this field without advice from Google. Not specifying the feild
```
Comment (slevenick) [inline mmv1/third_party/terraform/website/docs/r/compute_attached_disk.html.markdown:109]:
```suggestion
	this field without advice from Google. Not specifying the field
```

File: mmv1/third_party/terraform/website/docs/r/compute_attached_disk.html.markdown [hunk 4]
```diff
@@ -101,6 +101,18 @@ The following arguments are supported:
...
+	will allow the the server to assign the correct interface.
+
+	Possible values:
```
Comment (slevenick) [inline mmv1/third_party/terraform/website/docs/r/compute_attached_disk.html.markdown:112]: This is contained in the first line of this description, so we can skip one or the other

**Review by slevenick — CHANGES_REQUESTED** [review]
Typo in description, otherwise looks good!

**Pull request merged**